File: TestComponent/src/main/java/example/com/testcomponent/statics/TestStaticC.java
package example.com.testcomponent.statics;

import android.util.Log;

/**
 * Created by chengjian on 18-4-19.
 */

public class TestStaticC {
    public static int c = 23;
    private static final String TAG = "Static_TestC";

    static {
        Log.d(TAG, "static C initializer: ");
        c = 11;
    }


}

File: TestComponent/src/main/java/example/com/testcomponent/statics/TestStaticC.kt
package example.com.testcomponent.statics

import android.util.Log

/**
 * Created by chengjian on 18-4-19.
 */

object TestStaticC {
    var c = 23
    private val TAG = "Static_TestC"

    //    public TestStaticC() {
    //
    //    }

//    init {
//        val testStaticC = TestStaticB()
//        Log.d(TAG, "static C initializer: ")
//        c = 11
//    }
    fun testPub(testStaticD: TestStaticD) {
        val testStaticC = TestStaticC()
    }

}
